Helper/IpAddress.php

```diff
@@ -0,0 +1,133 @@
+<?php
+/**
+ *                       ######
+ *                       ######
+ * ############    ####( ######  #####. ######  ############   ############
+ * #############  #####( ######  #####. ######  #############  #############
+ *        ######  #####( ######  #####. ######  #####  ######  #####  ######
+ * ###### ######  #####( ######  #####. ######  #####  #####   #####  ######
+ * ###### ######  #####( ######  #####. ######  #####          #####  ######
+ * #############  #############  #############  #############  #####  ######
+ *  ############   ############  #############   ############  #####  ######
+ *                                      ######
+ *                               #############
+ *                               ############
+ *
+ * Adyen Payment module (https://www.adyen.com/)
+ *
+ * Copyright (c) 2020 Adyen BV (https://www.adyen.com/)
+ * See LICENSE.txt for license details.
+ *
+ * Author: Adyen <magento@adyen.com>
+ */
+
+namespace Adyen\Payment\Helper;
+
+use Adyen\Util\IpAddress as IpAddressUtil;
+use Magento\Framework\App\CacheInterface;
+use Magento\Framework\Serialize\SerializerInterface;
+
+/**
+ * Class IpAddress
+ * @package Adyen\Payment\Helper
+ */
+class IpAddress
+{
+
+    const IP_ADDRESS_CACHE_ID = "Adyen_ip_address";
+    const IP_ADDRESS_CACHE_LIFETIME = 86400;
+
+    /**
+     * @var IpAddressUtil
+     */
+    private $ipAddressUtil;
+
+    /**
+     * @var CacheInterface
+     */
+    private $cache;
+
+    /**
+     * @var SerializerInterface
+     */
+    private $serializer;
+
+    /**
+     * IpAddress constructor.
+     *
+     * @param IpAddressUtil $ipAddressUtil
+     * @param CacheInterface $cache
+     * @param SerializerInterface $serializer
+     */
+    public function __construct(
+        IpAddressUtil $ipAddressUtil,
+        CacheInterface $cache,
+        SerializerInterface $serializer
+    ) {
+        $this->ipAddressUtil = $ipAddressUtil;
+        $this->cache = $cache;
+        $this->serializer = $serializer;
+    }
+
+    /**
+     * Checks if the provided array of IPs addresses has been validated before and updates the cache accordingly
+     *
+     * @param string[] $ipAddresses
+     * @param bool $fullCacheKeyUpdate
+     * @return bool
+     */
+    public function isIpAddressValid($ipAddresses)
+    {
+        if (empty($ipAddresses)) {
+            return false;
+        }
+
+        $cachedIpsArray = $this->getIpAddressesFromCache();
```

Comment:
Add some validation here. If cache key is empty what can happen if the cron is not running. Let's retrieve it here. Maybe it is even better to remove the full cron and always do it here with a cache key that we define for one day.

Reply:
I added a log message to the checks on cron (cronjob.log) and helper (debug.log):

https://github.com/Adyen/adyen-magento2/commit/f6ed1dbc77052c1363451a5123f5e2d3169e08af#diff-28e83357bad28c013fdb1ead16fee88aR98

https://github.com/Adyen/adyen-magento2/commit/f6ed1dbc77052c1363451a5123f5e2d3169e08af#diff-71e336a11f03788f9a758eab93b6575eR60

Reply:
Added a cached IP addresses update on notification receiving @rikterbeek  https://github.com/Adyen/adyen-magento2/pull/713/files#diff-28e83357bad28c013fdb1ead16fee88aR100